lib/screens/home/widgets/category_main_menu.dart

```diff
@@ -15,33 +23,106 @@ class _CategoryMainMenuState extends ConsumerState<CategoryMainMenu> {
         .cocktailMenuTiles
-        .map((tile) => _buildGridTile(withName: tile.name))
+        .map((tile) => _buildGridTile(menu: tile))
         .toList();
 
-    return Center(
-        child: GridView.count(
-      primary: false,
-      padding: const EdgeInsets.all(20),
-      crossAxisSpacing: 10,
-      mainAxisSpacing: 10,
-      crossAxisCount: 2,
-      children: gridTile,
-    ));
+    return Container(
+        decoration: const BoxDecoration(
+          color: Color(AppColors.background),
+        ),
+        child: CustomScrollView(
+          slivers: [
+            SliverAppBar(
+              expandedHeight: 440,
+              backgroundColor: const Color(AppColors.green),
+              flexibleSpace: FlexibleSpaceBar(
+                background: ClipRRect(
+                  borderRadius: const BorderRadius.only(
+                    bottomLeft: Radius.circular(50),
+                    bottomRight: Radius.circular(50),
+                  ),
+                  child: ImageFiltered(
+                    imageFilter: ColorFilter.mode(
+                        const Color(AppColors.imageFilter).withOpacity(0.7),
+                        BlendMode.darken),
+                    child: Image.asset(
+                      'assets/background.png',
+                      fit: BoxFit.cover,
+                    ),
+                  ),
+                ),
+                title: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: const [
+                    Text("Cocktails app", textAlign: TextAlign.center),
+                  ],
+                ),
+              ),
+              shape: const RoundedRectangleBorder(
+                borderRadius: BorderRadius.only(
+                  bottomLeft: Radius.circular(50),
+                  bottomRight: Radius.circular(50),
+                ),
+              ),
+            ),
+            SliverPadding(
+              padding: const EdgeInsets.all(16),
+              sliver: SliverGrid(
+                gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                  crossAxisCount: 2,
+                  crossAxisSpacing: 10,
+                  mainAxisSpacing: 10,
+                ),
+                delegate: SliverChildListDelegate(
+                  gridTile,
+                ),
+              ),
+            ),
+          ],
+        ));
   }
 
-  Widget _buildGridTile({required String withName}) {
-    return GridTile(
-      child: Container(
-        padding: const EdgeInsets.all(8),
-        decoration: BoxDecoration(
-            border: Border.all(color: Colors.amberAccent, width: 5),
-            color: Colors.amber.shade100,
-            borderRadius: const BorderRadius.all(Radius.circular(20))),
-        child: Center(
-            child: Text(
-          withName,
-          textAlign: TextAlign.center,
-        )),
+  Widget _buildGridTile({required CocktailMenu menu}) {
```

Comment:
Instead of functions, we can improve the performance of the UI by building separate widgets. My bad to add the sample with the `_buildWidgetFunctions` which is usually helpful but on huge _buildWidgetFunctions I would prefer a new widget definition